crates/turborepo-lib/src/task_graph/visitor.rs

```diff
@@ -838,21 +860,49 @@ impl ExecContext {
 
         match exit_status {
             ChildExit::Finished(Some(0)) => {
+                // Attempt to flush stdout_writer and log any errors encountered
                 if let Err(e) = stdout_writer.flush() {
                     error!("{e}");
-                } else if let Err(e) = self
-                    .task_cache
-                    .save_outputs(&mut prefixed_ui, task_duration, telemetry)
-                    .await
-                {
-                    error!("error caching output: {e}");
                 } else {
-                    self.hash_tracker.insert_expanded_outputs(
-                        self.task_id.clone(),
-                        self.task_cache.expanded_outputs().to_vec(),
-                    );
+                    let mut cacheable = true;
+                    // If stdout_writer flush is successful, proceed with task processing
+                    if self.task_access.is_enabled() {
+                        // Task access enabled: Handle trace file processing
+                        if let Some(trace) =
+                            TaskAccessTraceFile::read(&self.task_access.repo_root, &self.task_hash)
+                        {
+                            // Trace file found
+                            if trace.can_cache(&self.task_access.repo_root) {
+                                // Cache trace and save task outputs if trace is cacheable
+                                self.task_access
+                                    .save_trace(trace, self.task_id_for_display.to_string());
```

Comment:
Just making sure my understanding is correct: we want this to ensure we're getting a valid `TaskName` that can appear in a pipeline definition so we're using `task_id_for_display` as it'll end up being a valid pipeline task name?

Reply:
Yep that's correct, a little weird in this case, but we need that to build the pipeline later on.